tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/CachedMessageHeadersHelper.cs

```diff
@@ -9,17 +9,25 @@
 using System.IO;
 using System.Reflection;
 using System.Reflection.Emit;
+using System.Text;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Shared;
 using Datadog.Trace.DuckTyping;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNS
 {
-    internal static class CachedMessageHeadersHelper<TMarkerType>
+    internal class CachedMessageHeadersHelper<TMarkerType> : IMessageHeadersHelper
```

Comment:
Making classes `sealed` can have performance benefits if we know there will never be a derived class.
```suggestion
    internal sealed class CachedMessageHeadersHelper<TMarkerType> : IMessageHeadersHelper
```